core/dbt/tracking.py

```diff
@@ -470,9 +435,51 @@ def process(self, record):
             )
 
 
-def initialize_from_flags():
+def initialize_from_flags(flags):
     # Setting these used to be in UserConfig, but had to be moved here
+    global active_user
     if flags.SEND_ANONYMOUS_USAGE_STATS:
-        initialize_tracking(flags.PROFILES_DIR)
+        active_user = User(flags.PROFILES_DIR)
+        try:
+            active_user.initialize()
+        except Exception:
+            fire_event(TrackingInitializeFailure())
+            active_user = User(None)
     else:
-        do_not_track()
+        active_user = User(None)
+
+
+@contextmanager
+def track_run(project_id, credentials, run_command):
```

Comment:
So, I think in the long run we'll want to take a little of the logic out of this function.  I don't think we want to require adapter data or profile name in order to start `track_run`, instead I think we need to start with nothing but the `flags` and `active_user` and then add another tracking event or two later on to ensure we still track that data.. I'm picturing something like this:

_for this PR_
```
@contextmanager
def track_run(run_command):
    invocation_context = get_base_invocation_context()

    track_invocation_start(invocation_context)
    try:
        yield
        track_invocation_end(invocation_context, result_type="ok")
    except (NotImplementedException, FailedToConnectException) as e:
        fire_event(MainEncounteredError(exc=str(e)))
        track_invocation_end(invocation_context, result_type="error")
    except Exception:
        track_invocation_end(invocation_context, result_type="error")
        raise
    finally:
        flush()
``` 

_for a future PR_
_gets called whenever we actually need the adapters_
```
def track_adapter_info(options):
    context = [SelfDescribingJson(ADAPTER_INFO, options)]
    assert active_user is not None, "Cannot track adapter info when active user is None"

    track(
        active_user,
        category="dbt",
        action="adapter_info",
        label=get_invocation_id(),
        context=context,
    )
```


Obvs, this will need to be done in tandem with changes to the internal analytics project/iglu schemas and is work outside the scope of _this_ ticket, but I wanted